rfcs/0102-moderation-team.md

```diff
@@ -0,0 +1,102 @@
+---
+feature: moderation team
+start-date: 2021-08-18
+author: tomberek
+co-authors: blaggacao
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/rfcs/pull/98
+---
+
+# Summary
+[summary]: #summary
+
+Establish a team to perform moderation.
+
+# Motivation
+[motivation]: #motivation
+
+There is not currently any official mechanism for moderation action. It's not
+sustainable to have to call on Graham any time there's a spammer or conflict
+that requires moderation, and we'd like to help the community become more
+self-regulating.
+
+The intention behind this RFC is to codify the current moderation practices.
+
+# Detailed design
+[design]: #detailed-design
+
+The Moderation Team is a volunteer group that may receive, invite, and evaluate
+applications to the team or alter the composition at any time. The team's
+composition, contact information, procedures, moderation log, and announcements
+should be available via
+[https://nixos.org/community/teams/moderation.html](https://nixos.org/community/teams/moderation.html).
+The team shall perform moderation activities on behalf of the community - with
+oversight via the RFC process and input from the NixOS Foundation - for
+discussions in [official project
+spaces](https://nixos.org/community/index.html) as well as unofficial spaces
+that seek and reach such an agreement with the team. The team should utilize
+the [NixOS Foundation mission](https://nixos.org/community/index.html) and the
+following statement during their duties:
+
+```
+The NixOS Foundation aims to promote participation without regard to gender,
+sexual orientation, disability, ethnicity, age, or similar personal
+characteristics. We want to strive to create and foster community by providing
+an intentionally welcoming and safe environment where all feel valued and cared
+for, and where all are given opportunity to participate meaningfully. The
+Foundation will work with the community in service of this goal.
+```
+
+ref: [twitter](https://twitter.com/grhmc/status/1390775249424338944)
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+- The initial Moderation Team - drawn from the existing Discourse and GitHub
+  teams - is defined to be @grahamc, @zimbatm, @domenkozar, @Mic92, @garbas,
+  and @ryantm.
+- Rename the Discourse Team to Moderation Team on
+  https://nixos.org/community/teams/discourse.html and utilize
+  https://nixos.org/community/teams/moderation.html.
+- Establish and publish a clear point of contact for abuse reporting and a
+  venue for discussion about moderation specific topics such as a dedicated
+  Matrix channel or Discourse topic.
+
```

Comment:
This RFC is missing some major parts of running a moderation team. Namely, it's missing the lifecycle management and onboarding of new moderators, and the phasing out of existing ones.

I understand that the primary purpose here is to ratify the status quo, but to be successful we also need to define the next steps and what defines a healthy moderation team.

A healthy moderation team often operates with a term based system - both to reduce burnout and ensure that nobody ends up in a position of being a moderator for life.

To that end, I think this RFC needs:
- an initial set of working practices (_that can be adjusted in the future_), as an example, the CoCC bootstrapping done within the kubernetes community https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/bootstrapping-process.md.
- to require an _odd_ number of members to prevent split-decisions when a vote is required (there are currently 6 suggested initial members and simplify quorum).
- a requirement that a formal charter (e.g https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/charter.md) is defined within some period after the initial committee is bootstrapped.

Reply:
In general I am not opposed to adding more detail, but we don't seem to differ too much from the k8s document, and we have more detail about the specifics.

- initial working practices are sparse, but do specify overall guiding principles that don't seem to be controversial (mission + twitter statement). Perhaps we add a few explicit expectations into the "Examples and Interactions" section? Though I don't quite know what we should add. Do you have a minimal recommended addition?

- odd number of members: We're trying to enumerate the current moderators (either de facto or de jure). I'd like to avoid adding or removing arbitrarily (and there is almost never a timezone or opportunity for all to meet anyway and deadlock can be seen as both a feature or a curse). I'm open to adding someone I may have missed.

- formal charter: I included some items in Future Work and in the statement "The team's composition, contact information, procedures, moderation log, and announcements should...". Add "charter" to that list? or change procedures to charter?

Reply:
> This RFC is missing some major parts of running a moderation team.

Yes, and these are currently handled pretty ad-hoc, and that will have to do for some more time.

RFC SC also seems to have some regular practices beyond what is written down as a public document, and moderation currently happens somehow, so formalising initial procedures seems optional based on experience.

It is unclear why do we need to make deadlocks on a decision procedurally impossible. After all, RFC SC needs to make a lot of decisions unanimously and progress still happens. Full team consultations take a lot of time either way so they cannot be a way to make instant decisions. And unlike RFC SC duties, quick decisions are a part of moderation.

We have already had an «X must be defined by Y» attempt (documentation system choice). I think that outcome is «the attempt failed, then a more typical RFC process happened, and took quite some time to converge». (I don't think the implementation has converged yet, either). I do not see any reasons to believe a deadline would bring us much good here.

RFC SC rotation rate is quite limited by new applications, and moderation needs more of trust in each member (for RFC SC with its unanimity procedures it is enough to have trust in good faith of all members and values of a half — especially when different project participants trust different subsets). So it's better to first just stress explicit rotation decisions as a part of moderation team duties, which this proposal does.

Overall I think each of the proposed additions would need non-trivial amount of discussion to decide in favour or against, and I see no argument why this has to be a part of the scope here.